source/extensions/filters/http/lua/lua_filter.cc

```diff
@@ -779,6 +779,17 @@ int StreamHandleWrapper::luaTimestampString(lua_State* state) {
   return 1;
 }
 
+int StreamHandleWrapper::luaNodeMetadata(lua_State* state) {
+  ASSERT(state_ == State::Running);
+  if (node_metadata_wrapper_.get() != nullptr) {
+    node_metadata_wrapper_.pushStack();
+  } else {
+    node_metadata_wrapper_.reset(
+        Filters::Common::Lua::MetadataMapWrapper::create(state, filter_.nodeMetadata()), true);
+  }
+  return 1;
+}
```

Comment:
should we extend the original metadata() method to take a optional parameter? A new method may have better performance but extend the exist method could re-use more code.

cc @mattklein123

Reply:
I think we should be more careful here because this will change the API. Waiting for input from @mattklein123 

/wait

Reply:
friendly ping @mattklein123